### PR TITLE
Allow protected apply methods

### PR DESCRIPTION
**Pull request by sebastiandedeyne**
`app()->call()` currently fails when a method is `protected`. 

`method_exists` returns `true` for protected methods here (probably because the check happens in the parent `AggregateRoot` class) When the protected methods get called, there's an exception because `app()->call()` wants a public method.

### CODE REVIEW

**Comment by freekmurze** [comment]
Nice catch!

Could you also add a test?

File: src/AggregateRoots/AggregateRoot.php [hunk 1]
```diff
@@ -195,12 +195,19 @@ private function apply(ShouldBeStored $event): void
...
 
-        if (method_exists($this, $applyingMethodName)) {
+        $reflectionClass = new ReflectionClass($this);
+
+        $applyMethodExists = $reflectionClass->hasMethod($applyingMethodName);
+        $applyMethodIsPublic = $applyMethodExists && $reflectionClass->getMethod($applyingMethodName)->isPublic();
+
+        if ($applyMethodExists && $applyMethodIsPublic) {
             try {
                 app()->call([$this, $applyingMethodName], ['event' => $event]);
```
Comment (riasvdv) [inline src/AggregateRoots/AggregateRoot.php:205]: What's the benefit of using `app()->call()`? Can't we just always do `$this->$applyingMethodName($event);` instead?

Comment (sebastiandedeyne) [reply src/AggregateRoots/AggregateRoot.php:205]: `app()->call()` allows the package user to inject additional dependencies in the `apply*` methods.

I don't see much value in that feature, but removing it now would be a breaking change.

Comment (riasvdv) [reply src/AggregateRoots/AggregateRoot.php:205]: Ah right, forgot about that

**Comment by sebastiandedeyne** [comment]
I added a test for the DI behaviour, which I believe we should reconsider in the next major version.

**Comment by sebastiandedeyne** [comment]
Rebased the branch to fix the tests. Anything else that needs to happen before we can merge?

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**